plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSModuleValidator.xtend

```diff
@@ -73,8 +74,11 @@ class N4JSModuleValidator extends AbstractN4JSDeclarativeValidator {
 
 	@Check
 	def void checkModuleSpecifier(Script script) {
+		val resType = ResourceType.getResourceType(script.module.eResource);
+		val correctType = resType === ResourceType.N4JS || resType === ResourceType.N4JSD || resType === ResourceType.N4JSX;
```

Comment:
maybe introduce helper like `org.eclipse.n4js.N4JSGlobals.ALL_N4_FILE_EXTENSIONS`

Reply:
good idea, but here we have ResourceTypes and N4JSGlobals is dealing with Strings